builder/processor/src/main/java/io/helidon/builder/processor/CustomMethods.java

```diff
@@ -0,0 +1,288 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.helidon.builder.processor;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+import io.helidon.common.Errors;
+import io.helidon.common.types.Annotation;
+import io.helidon.common.types.TypeInfo;
+import io.helidon.common.types.TypeName;
+
+import static io.helidon.builder.processor.Types.BUILDER_CUSTOM_METHOD_TYPE;
+import static io.helidon.builder.processor.Types.CUSTOM_METHODS_TYPE;
+import static io.helidon.builder.processor.Types.FACTORY_METHOD_TYPE;
+import static io.helidon.builder.processor.Types.PROTOTYPE_CUSTOM_METHOD_TYPE;
+import static io.helidon.builder.processor.Types.VOID_TYPE;
+
+record CustomMethods(List<CustomMethod> factoryMethods,
+                     List<CustomMethod> builderMethods,
+                     List<CustomMethod> prototypeMethods) {
+    static CustomMethods create(ProcessingContext ctx, TypeContext.TypeInformation typeInformation) {
+        Optional<Annotation> annotation = typeInformation.blueprintType().findAnnotation(CUSTOM_METHODS_TYPE);
+        if (annotation.isEmpty()) {
+            return new CustomMethods(List.of(), List.of(), List.of());
+        }
+        // value is mandatory for this annotation
+        String customMethodType = annotation.get().value().orElseThrow();
+        // we must get the type info, as otherwise this is an invalid declaration
+        TypeInfo customMethodsInfo = ctx.toTypeInfo(TypeName.create(customMethodType))
+                .orElseThrow();
+
+        Errors.Collector errors = Errors.collector();
+        List<CustomMethod> factoryMethods = findMethods(typeInformation,
+                                                        customMethodsInfo,
+                                                        errors,
+                                                        FACTORY_METHOD_TYPE,
+                                                        CustomMethods::factoryMethod);
+        List<CustomMethod> builderMethods = findMethods(typeInformation,
+                                                        customMethodsInfo,
+                                                        errors,
+                                                        BUILDER_CUSTOM_METHOD_TYPE,
+                                                        CustomMethods::builderMethod);
+        List<CustomMethod> prototypeMethods = findMethods(typeInformation,
+                                                          customMethodsInfo,
+                                                          errors,
+                                                          PROTOTYPE_CUSTOM_METHOD_TYPE,
+                                                          CustomMethods::prototypeMethod);
+
+        errors.collect().checkValid();
```

Comment:
this is an API that is with Helidon since version 1, there is no need to do this backward incompatible change based on naming concerns. You can do a PR to add a new method and deprecate this old one if you feel like it ;)